test/unit-test/twcc_manager/twcc_manager_utest.c

```diff
@@ -28,30 +112,167 @@ void test_twccAddPacket( void )
+                           result );
+        TEST_ASSERT_EQUAL( i + 1,
+                           twccManager.count );
+    }
+
+    twccPacketInfo.packetSize = packetSize + ( size_t ) i;
+    twccPacketInfo.localSentTime = sentTime + ( uint64_t ) i;
+    twccPacketInfo.packetSeqNum = seqNum + ( uint16_t ) i;
+
+    result = RtcpTwccManager_AddPacketInfo( &( twccManager ),
+                                            &( twccPacketInfo ) );
+
+    TEST_ASSERT_EQUAL( RTCP_TWCC_MANAGER_RESULT_OK,
+                       result );
+    TEST_ASSERT_EQUAL( TWCC_PACKET_INFO_ARRAY_LENGTH,
+                       twccManager.count );
+
+    for( i = 1; i <= TWCC_PACKET_INFO_ARRAY_LENGTH; i++ )
+    {
+        result = RtcpTwccManager_FindPacketInfo( &( twccManager ),
+                                                 seqNum + ( uint16_t ) i,
+                                                 &( foundTwccPacketInfo ) );
+
+        TEST_ASSERT_EQUAL( RTCP_TWCC_MANAGER_RESULT_OK,
+                           result );
+        TEST_ASSERT_EQUAL( sentTime + ( uint64_t ) i,
+                           foundTwccPacketInfo.localSentTime );
+        TEST_ASSERT_EQUAL( packetSize + ( size_t ) i,
+                           foundTwccPacketInfo.packetSize );
+        TEST_ASSERT_EQUAL( seqNum + ( uint16_t ) i,
+                           foundTwccPacketInfo.packetSeqNum );
+    }
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Validate Twcc Manager check packet deletion functionality when the added packet timestamp is less then any packet in the packetInfoArray.
+ */
+void test_twccOlderPacketInfoDeletion_AddOldPacket( void )
+{
+    uint16_t seqNum = 256;
+    RtcpTwccManager_t twccManager;
+    RtcpTwccManagerResult_t result;
+    TwccPacketInfo_t twccPacketInfo = { 0 };
+    TwccPacketInfo_t foundTwccPacketInfo;
+
+    result = RtcpTwccManager_Init( &( twccManager ),
+                                   &( twccPacketInfoArray[ 0 ] ),
+                                   TWCC_PACKET_INFO_ARRAY_LENGTH );
+
+    TEST_ASSERT_EQUAL( RTCP_TWCC_MANAGER_RESULT_OK,
+                       result );
+
+    twccPacketInfo.localSentTime = time( NULL );
+    twccPacketInfo.packetSize = 0;
+    twccPacketInfo.packetSeqNum += seqNum;
+
+    result = RtcpTwccManager_AddPacketInfo( &( twccManager ),
+                                            &( twccPacketInfo ) );
+    TEST_ASSERT_EQUAL( RTCP_TWCC_MANAGER_RESULT_OK,
+                       result );
+
+    twccPacketInfo.packetSize = 1;
+    twccPacketInfo.localSentTime -= RTCP_TWCC_ESTIMATOR_TIME_WINDOW;
+    twccPacketInfo.packetSeqNum += seqNum;
+
+    result = RtcpTwccManager_AddPacketInfo( &( twccManager ),
+                                            &( twccPacketInfo ) );
+    TEST_ASSERT_EQUAL( RTCP_TWCC_MANAGER_RESULT_OK,
+                       result );
+
+    /* The last added packet added will have a less time-stamp hence no packet will get deleted */
```

Comment:
Does this match any RFC description? Or should we simply drop the packet while adding if the local sent time is less?

Reply:
For this I have discussed with Gaurav yesterday, in actual cases TWCC packets will always come in sequence and have local sent time larger in each case. So whatever happens when local sent time is less, doesn't matter much. Either we can edit the source code to delete that packet. Or just add it. 

Since this scenerio won't happen in real life.